### PR TITLE
Shorten no-referral GitHub URLs too

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes refined-github/refined-github#4981

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -438,4 +438,12 @@ test('GitHub.com URLs', urlMatcherMacro, new Map([
 		'https://github.com/sindresorhus/notifier-for-github/pull/253/files/6b4489d417c9425dc27c5fb8d6b4a8518debd035..60cdcf3c3646164441bf8f037cef620479cdec59',
 		'<code>6b4489d4..60cdcf3c</code> (#253)',
 	],
+	[
+		'https://togithub.com/fregante/shorten-repo-url/pull/32/files',
+		'#32 (files)',
+	],
```
Comment (fregante) [inline test.js:444]: Does this work too?
```suggestion
	[
		'https://togithub.com/fregante/shorten-repo-url/pull/32',
		'#32',
	],
	[
		'https://togithub.com/fregante/shorten-repo-url/pull/32/files',
		'#32 (files)',
	],
```

**Comment by fregante** [comment]
Yeah it doesn't because those links are usually shortened natively. To properly add support for https://github.com/refined-github/refined-github/issues/4981 the regular issues should also work. The example URL in that issue isn't shortened

**Review by fregante — APPROVED** [review]
Probably good 😃

**Comment by cheap-glitch** [comment]
Are there any other URLs that are natively shortened, besides basic issue and PR links?

File: test.js [hunk 2]
```diff
@@ -446,6 +450,10 @@ test('GitHub.com URLs', urlMatcherMacro, new Map([
 		'https://togithub.com/fregante/shorten-repo-url/pull/32/files',
 		'#32 (files)',
 	],
+	[
+		'https://togithub.com/nodejs/node/pull/123/files',
+		'nodejs/node#123',
```
Comment (cheap-glitch) [inline test.js]: This should have failed but it didn't… There's probably a bug lurking somewhere.

Comment (cheap-glitch) [reply test.js]: Never mind, forgot this was a `Map` and thus values with the same keys are overwritten :sweat_smile:

File: index.js [hunk 3]
```diff
@@ -179,6 +188,21 @@ function shortenURL(href, currentUrl = 'https://github.com') {
...
+		if (issue) {
+			return `${repoUrl}#${issue}`;
+		}
+
+		if (pull) {
+			return `${repoUrl}#${pull}`;
+		}
+
+		if (commit) {
+			return `<code>${commit.slice(0, 7)}</code>`;
```
Comment (fregante) [inline index.js:202]: This doesn't support cross-repo commit links, like

https://github.com/refined-github/refined-github/commit/4f270c4f50e0a2a20085a6e92095117f10340322

**Review by fregante — COMMENTED** [review]
Ideally this should support issues and PRs even without `isRedirection`, to both work as a general-purpose shortener (rather than extension-specific)

**Review by fregante — APPROVED** [review]
<3

File: index.js [hunk 4]
```diff
@@ -4,7 +4,9 @@ const patchDiffRegex = /[.](patch|diff)$/;
 const releaseRegex = /^releases[/]tag[/]([^/]+)/;
 const labelRegex = /^labels[/]([^/]+)/;
 const compareRegex = /^compare[/]([^/]+)/;
-const pullRegex = /^pull[/](\d+)[/]([^/]+)(?:[/]([\da-f]{40})[.][.]([\da-f]{40}))?$/;
+const pullRegex = /^pull[/](\d+)(?:[/]([^/]+))?(?:[/]([\da-f]{40})[.][.]([\da-f]{40}))?$/;
```
Comment (fregante) [inline index.js:7]: It'd be good to use named groups here to describe what each group is capturing, even if unused. What do you think?

**Comment by fregante** [comment]
You can open a PR with the same title on refined GitHub and install just this dependency

File: index.js (hunk 4 above)
Comment (cheap-glitch) [reply index.js:7]: Yes :+1:

**Pull request merged**